backend/prisma/schema.prisma

```diff
@@ -7,31 +7,110 @@ generator prisma_client {
   provider = "prisma-client-js"
 }
 
-enum Status {
-  ADMIN MODERATOR FREE_USER PRO_USER
+model User {
+  Id                String        @id @default(cuid())
+  Name              String
+  Email             String        @unique
+  Password          String
+  ResetToken        String?
+  ResetTokenExpiry  Float?
+  userRole          UserRole      @default(FREE_USER)
+  CreatedAt         DateTime      @default(now())
+  UpdatedAt         DateTime      @updatedAt
+  Handle            String        @unique
+  Bio               String
+  SocialMedia       SocialMedia[]
+  interests         Topic[]
+  location          Location?
+  languagesNative   Language[]    @relation("LanguageToNative")
+  languagesLearning Language[]    @relation("LanguageToLearner")
 }
 
-model User {
-  Id               String   @id @default(cuid())
-  Name             String
-  Email            String   @unique
-  Password         String
-  ResetToken       String?
-  ResetTokenExpiry Float?
-  status           Status[]
-  posts            Post[]
-  CreatedAt        DateTime @default(now())
-  UpdatedAt        DateTime @updatedAt
+model SocialMedia {
+  Id       String              @id @default(cuid())
+  Platform SocialMediaPlatform
 }
 
 model Post {
-  Id         String   @id @default(cuid())
+  Id         String     @id @default(cuid())
   Title      String
   Body       String
-  Published  Boolean
+  status     PostStatus @default(DRAFT)
   Image      String?
   LargeImage String?
-  CreatedAt  DateTime @default(now())
-  UpdatedAt  DateTime @updatedAt
+  CreatedAt  DateTime   @default(now())
+  UpdatedAt  DateTime   @updatedAt
   author     User
+  ReadCount  Int        @default(0)
+  ReadTime   Int
+  likes      Like[]
+  comments   Comment[]
+  tags       Tag[]
+  topic      Topic[]
+  language   Language
```

Comment:
I just had a thought... what if people want to write posts in multiple languages?? Maybe something we could add support for later if people asked for it. I'm thinkin' we stick to simple and see how it goes/what users ask for

Reply:
Yeah I already thought about this and I actually think that it is a constraint I would like to keep for the foreseeable future. I don't think we need to support people writing a post in multiple languages -- it could add so many complications.

One of the strengths of Journaly architecturally is how simple and well-structured the data is going to be, which will be so helpful for building powerful search and matching capabilities and incorporating ML, for example. Posts are organized by languages and topics. Imagine how complex it could become both for users and for our tooling if suddenly posts can be in multiple languages.

For example, our future amazing tools for running realtime text analysis as a user types to help them improve their writing in French. Or users who speak English and are learning German and so have their personal feed all nicely set up to only show them those posts, but suddenly start encountering posts where they can only understand or provide feedback on one paragraph in the middle. I thought of a bunch more examples we can chat about.

Personally, I think it's totally fair for those users who want to write in multiple languages to simply have to write posts in each one 😊 We could provide cool features later that help them, such as a "translation" tool where they can write a post in their native language (or just paste some existing text they like) and then practice translating it into a language they're learning to get feedback, and they could do this for several languages if they wanted.

But anyway, those are my thoughts on this 🙂

Reply:
Those are definitely good points! If we for some reason did want to allow people to write in multiple languages without complicating the DB structure, we could actually have a language called "all" or "world" or "multiple". I've seen that strategy used for entities that are available everywhere instead of one country locale. Sounds good though to just keep it simple for now!